### DISCUSSION TITLE
How do you disconnect your bot after some time of inactivity?

### DISCUSSION THREAD

**Discussion by forbidmellm** [discord-jda/JDA#1886] [Questions and Help] [answered]
I don't know how to do this correctly

**Comment by Yashar256** [comment, marked as answer]
I don't think there is direct support for this, but it is possible if you know how to [multithread](https://www.tutorialspoint.com/java/java_multithreading.htm). Whenever someone uses a bot command, you can set an [AtomicLong](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/atomic/AtomicLong.html) to the timestamp it was last used at, and then have another thread that checks if its been a while since a command was used every so often (You can use a [ScheuledExectuorService](https://docs.oracle.com/javase/8/docs/api/java/util/concurrent/ScheduledExecutorService.html) to make it easier to check).

**Comment by DV8FromTheWorld** [comment]
When you say "disconnect your bot", what do you mean?
Do you mean leave an audio channel if there is no one in it listening? Turn off the bot if it hasn't seen a message within X minutes? Leave a guild if it hasn't handled a command from a user in the last month from that guild?

You need to provide more information.
